application/src/main/java/org/thingsboard/server/actors/calculatedField/CalculatedFieldEntityMessageProcessor.java

```diff
@@ -197,12 +197,18 @@ private void processArgumentValuesUpdate(CalculatedFieldCtx ctx, List<Calculated
             state = getOrInitState(ctx);
             justRestored = true;
         }
-        if (state.updateState(newArgValues) || justRestored) {
-            cfIdList = new ArrayList<>(cfIdList);
-            cfIdList.add(ctx.getCfId());
-            processStateIfReady(ctx, cfIdList, state, tbMsgId, tbMsgType, callback);
-        } else {
-            callback.onSuccess(CALLBACKS_PER_CF);
+        try {
+            if (state.updateState(newArgValues) || justRestored) {
+                cfIdList = new ArrayList<>(cfIdList);
+                cfIdList.add(ctx.getCfId());
+                processStateIfReady(ctx, cfIdList, state, tbMsgId, tbMsgType, callback);
+            } else {
+                callback.onSuccess(CALLBACKS_PER_CF);
+            }
+        } catch (Exception e) {
+            if (DebugModeUtil.isDebugFailuresAvailable(ctx.getCalculatedField())) {
```

Comment:
need to throw exception of execute `callback.onFailure`

Reply:
I don't think it is a good practice to try catch every method and log debug events there. If you need to try-catch - it is ok. But instead of logging debug in every method, re-throw the new exception and catch it in the 
`CalculatedFieldEntityActor.doProcess` or on a bit lower level, but not in every method.